index.ts

```diff
@@ -1,17 +1,55 @@
 type SmartlookWindow = Window & { smartlook?: any }
 const SL_NOT_INITIALIZED = 'Smartlook client is not initialized.'
 
+interface Network {
+	allowedHeaders?: string[]
+	allowedUrls?: (string | RegExp)[]
+}
+
+type RequestOrResponse = {
+	body?: string
+	headers?: Record<string, string[]>
+}
+
+interface Interceptors {
+	network?: (
+		data: {
+			request?: RequestOrResponse
+			response?: RequestOrResponse
+			url: string
+		},
+		context: { pageUrl: string },
+	) => void | boolean
+	url?: (data: { url: string }) => void
+}
+
+type InitParams = {
+	// Advanced network recording allows recording the bodies and headers of requests and responses.
```

Comment:
these descriptions should go to README.md as well... ideally add there also links to appropriate pages to our docs.